t/alignment/array.xml

```diff
@@ -5,7 +5,12 @@
 <document xmlns="http://dlmf.nist.gov/LaTeXML">
   <resource src="LaTeXML.css" type="text/css"/>
   <resource src="ltx-article.css" type="text/css"/>
-  <section refnum="1" xml:id="S1">
+  <section xml:id="S1">
+    <tags>
+      <tag>1</tag>
+      <tag role="refnum">1</tag>
+      <tag role="typerefnum">§1</tag>
```

Comment:
why not `trefnum` for a role, given there is `frefnum`? I am unsure if it's better to argue for a single letter convention, or longer attribute names such as `refnum_formatted` and `refnum_type` or some such.

Also curious to observe that they really have the same refnum *role*, but hold distinct `variant`s for that role. Though I suspect adding another attribute is overhead that's not its time right now.

Reply:
I spent a lot of time, and back-and-forth, with the names, trying to improve over "frefnum" :>  There are 3 base cases roughly equivalent to the former `frefnum`, `refnum` and `rrefnum`, but hopefully with more explanatory names, but perhaps those names can be further clarified.

* No role is a "formatted refnum"; it is used for the display in or alongside the object being tagged (eg. items, equations) Exception is that objects with a title or caption, the tag is essentially duplicated within the title/caption so the one in the `ltx:tags` is likely never used.
* `@role=refnum` is the plain refnum, essentially `\thecounter` which LaTeX would use to fill in `\ref`.  Note that the author typically gives some context like `see section \ref{foo}`, and that not all objects get a refnum, in which case latex uses the refnum of the parent.
* `@role=typerefnum` is a "typed refnum" for use when generating references to an object, typically automated, such as back-references and such. There's no context there, so the type name or symbol is helpful.

Hope that helps; if you can think of better names...

Reply:
Thanks! Got it.

Well, `refnum_formatted` and `refnum_typed` do sound better to me, but they are also long, so probably not ideal. I still slightly lean towards recommending them...

Reply:
The other, more informal-sounding, bunch of names that come to mind are things like `fmtnum` and `typnum`, but I don't really find the appeal in such dashing shorthands nowadays.